enhancements/network/core-network-policies.md

```diff
@@ -0,0 +1,547 @@
+
+
+#### Removing Policy Restrictions for OpenShift
+
+1. The Cluster Administrator can deploy an AdminNetworkPolicy we
+   will document with a public KCS and test
+
+2. The policy will bypass Network Policy for all namespaces with the
+   label described above so that ingress and egress traffic will succeed
+   without restriction (this is the current behavior)
+
+3. This is only intended to be an escape hatch that support might need
+   to deploy until an appropriate targeted policy can be applied
+
+
+#### Developing and Shipping Network Policy
+
+1. OpenShift developers and Third-Party developers will both need to
+   create network policies for the pods running in their namespaces
+
+2. Developing policies:
+
+    1. Document what traffic flows (ingress and egress) need to be
+       allowed for pods in the namespaces
+    
+    2. The ACS product can be used to analyze workloads in a cluster to
+    
+        1. understand which connections have live traffic (active connections)
+        
+        2. understand which connections have no traffic (hinting that
+           these connections should be blocked by network policy)
+        
+        3. Suggest (generate) a network policy based on the observed traffic baseline
+        
+    3. ACS can also be used to analyze a folder with all of the
+       namespace YAML resources to heuristically try to figure out who is
+       trying to talk to whom and mint tight network policies to reflect
+       that model.
+    
+    4. The Network Observability tool can be used to detect when a network policy blocked traffic
+    
+3. Auditing policies:
+
+    5. Someone familiar with network policies should work with the
+       team to ensure that the policies are as tight as is possible given
+       the expected traffic
+    
+4. Deploying policies:
+
+    6. The network policies need to be deployed by the operator in to the operand namespaces
+
+
+### API Extensions
+
+This extends the API server by changing the Admission Controller on
+namespace objects so that the label `security.openshift.io/openshift-namespace` is applied
+or removed depending on whether the namespace name starts with
+`openshift-`.
+
+Other than that, there is no other API change.
+
+
+### Topology Considerations
+
+
+#### Hypershift / Hosted Control Planes
+
+Are there any unique considerations for making this change work with Hypershift?
+
+No. 
+
+
+#### Standalone Clusters
+
+Is the change relevant for standalone clusters?
+
+No.
+
+
+#### Single-node Deployments or MicroShift
+
+How does this proposal affect the resource consumption of a
+single-node OpenShift deployment (SNO), CPU and memory?
+
+No change, other than the creation of some additional objects in the API server.
+
+How does this proposal affect MicroShift? For example, if the proposal
+adds configuration options through API resources, should any of those
+behaviors also be exposed to MicroShift admins through the
+configuration file for MicroShift?
```

Comment:
Microshift doesn't run the cluster-network-operator.  How will upgrades of microshift get the appropriate namespace labels?  How will upgrades prevent non-openshift-* namespaces from squatting on the label?